src/RamExecutor.cpp

```diff
@@ -2511,11 +2625,11 @@ std::string RamCompiler::generateCode(
 }
 
 std::string RamCompiler::compileToLibrary(
-        const SymbolTable& symTable, const RamProgram& prog, const std::string& filename) const {
-    std::string source = generateCode(symTable, prog, filename + ".cpp");
+        const SymbolTable& symTable, const RamProgram& prog, const std::string& filename, const int index) const {
+    std::string source = generateCode(symTable, prog, filename, index);
```

Comment:
Could you rename `source` to, say, `sourceFilename`? It's particularly unclear when it's assigned to be a method named `generateCode`.